### ISSUE TITLE
Unable to evaluate ... method threw LazyInitializerException

### ISSUE THREAD

**Issue by zurfyx** [UdL-EPS-SoftArch/softarch-1617-server#37]
**Ocurring in:**
https://github.com/UdL-EPS-SoftArch/softarch-1617-server/blob/f170a420928150be646fd3454e02082f41f4c37e/src/test/java/cat/udl/eps/softarch/steps/PurchaseStepDefs.java#L51

**When:**
Trying to retrieve (serialize) fields that are OneToOne, ManyToOne, Collections, and related.
```
Advertisement advertisement = new Advertisement();
Purchase purchase = new Purchase();
purchase.setAdvertisement(advertisement);
String message = mapper.writeValueAsString(purchase); <-- LazyInitializerException
```

**Related:**
http://stackoverflow.com/questions/13671178/org-hibernate-lazyinitializationexception-could-not-initialize-proxy-no-sessi
http://stackoverflow.com/questions/27299235/spring-mvc-hibernate-could-not-initialize-proxy-no-session

**Comment by rogargon** [comment]
Are these entities persisted using the corresponding repositories? Otherwise they don't have ids and the serialisation cannot be generated. Consider creating the Advertisement in a separate Given step.

**Comment by rogargon** [comment]
Fixed with commits 4ae16cf21501c74615cc354c35155077fba2aa0a and 061b407332962722526a8d72c19899edb690f963
